keps/sig-auth/2718-20210511-client-exec-proxy/README.md

```diff
@@ -0,0 +1,837 @@
+proxy server to be executed and communicated with according to configuration
+solely contained in a kubeconfig file.
+3. Defining a specification which is interoperable with an existing client-side
+proxy setup, e.g. a proxy configured with the HTTP_PROXY environment variable.
+4. Defining a specification for a client-side proxy server that is easy to
+secure without taking additional setup steps, by:
+    - Using a unix domain socket for the communication between the client and
+      the proxy, secured with file permissions and TLS.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+The following are non-goals of this KEP:
+1. Solving use cases outside of the above-mentioned client authentication
+   workflows.
+2. Implementing a specific flavor of request signing (like SigV4), or adding
+   support for any hardware security module protocol directly in the Kubernetes
+   client libraries.
+3. Defining any new mechanisms for validating request signatures server-side.
+   This is because they can be validated by a front-proxy, which is already a
+   feature of Kubernetes.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+This proposal is to implement an extension in Kubernetes client libraries to
+forward requests to a proxy process for request modification.  The proxy will
+receive an HTTP request and will have the opportunity to sign the request,
+attach headers, use hardware protected keys for mTLS, and/or modify the
+request.  The final request will be constructed and forwarded to the apiserver.
+
+It is already possible, by setting the appropriate HTTP_PROXY, HTTPS_PROXY
+environment variables, to configure a Kubernetes client to send requests to a
+proxy.  While this may work in some situations, it is not ideal for a generic
+request signing solution across all client configurations.  Firstly, http
+proxies are already commonly integrated in client networking configurations, so
+for these clients setting up an additional proxy configuration might not be
+possible, or at least not straightforward.  Secondly, this approach requires a
+long running external proxy process which introduces operational complexity.
+Thirdly, communicating with a proxy for request signing requires a secure
+communication channel between the client and the proxy, which means in a case
+where multiple users share a client computer (for example, a shared bastion),
+configuring and securing a local proxy to prevent signing requests as a
+different user becomes more complicated.
+
+### Risks and Mitigations
```

Comment:
IMO we do not need to do anything because we do not rely on the UDS to provide any security properties.  https://github.com/kubernetes/enhancements/pull/2693#discussion_r986893116